pkg/http/oidc_authenticator.go

```diff
@@ -254,22 +254,32 @@ func (a *oidcAuthenticator) Authenticate(w http.ResponseWriter, r *http.Request)
 		}
 	}
 
-	// No valid session present. Redirect to the authorization service.
-	var stateVerifier [16]byte
-	if _, err := a.randomNumberGenerator.Read(stateVerifier[:]); err != nil {
-		return nil, err
-	}
-	if err := a.setCookieValue(w, &oidc.CookieValue{
-		SessionState: &oidc.CookieValue_Authenticating_{
-			Authenticating: &oidc.CookieValue_Authenticating{
-				StateVerifier:      stateVerifier[:],
-				OriginalRequestUri: r.URL.RequestURI(),
+	// No valid session present. Redirect to the authorization service if this is a top level page navigation (or unknown).
+	// Otherwise, return a 401 Unauthorized.
+
+	fetchType := r.Header.Get("Sec-Fetch-Dest")
+
+	if fetchType == "document" || fetchType == "" {
```

Comment:
Could you please write something along the following lines instead?

```go
if fetchType := r.Header.Get("Sec-Fetch-Dest"); fetchType != "" && fetchType != "document" {
    return nil, status.Error(codes.Unauthenticated, "No valid OIDC session state cookie found")
}
```

We already have logic in place to convert such errors to HTTP 401.